r/vignettes/arrow.Rmd

```diff
@@ -1,227 +1,222 @@
 
-In the tables, entries with a `-` are not currently implemented.
+To learn more about the different data object classes in `arrow`, see the article on [data objects](./data_objects.html).
 
-### R to Arrow
+## Converting Tables to data frames
 
-| R type                   | Arrow type |
-|--------------------------|------------|
-| logical                  | boolean    |
-| integer                  | int32      |
-| double ("numeric")       | float64^1^ |
-| character                | utf8^2^    |
-| factor                   | dictionary |
-| raw                      | uint8      |
-| Date                     | date32     |
-| POSIXct                  | timestamp  |
-| POSIXlt                  | struct     |
-| data.frame               | struct     |
-| list^3^                  | list       |
-| bit64::integer64         | int64      |
-| hms::hms                 | time32     |
-| difftime                 | duration   |
-| vctrs::vctrs_unspecified | null       |
+Tables are a data structure used to represent rectangular data within memory allocated by the Arrow C++ library, but they can be coerced to native R data frames (or tibbles) using `as.data.frame()`
 
+```{r}
+as.data.frame(dat)
+```
+
+When this coercion takes place, each of the columns in the original Arrow Table must be converted to native R data objects. In the `dat` Table, for instance, `dat$x` is stored as the Arrow data type int32 inherited from C++, which becomes an R integer type when `as.data.frame()` is called. 
+In most instances the data conversion takes place automatically and without friction: a column stored as a timestamp in Arrow becomes a POSIXct vector in R, for example. However, there are some instances where the mapping between Arrow data types and R data types is not exact and care is required.
 
+To learn more about data types in Arrow and how they are mapped to R data types, see the [data types](./data_types.html) article. 
 
-^1^: `float64` and `double` are the same concept and data type in Arrow C++; 
-however, only `float64()` is used in arrow as the function `double()` already 
-exists in base R
 
-^2^: If the character vector exceeds 2GB of strings, it will be converted to a 
-`large_utf8` Arrow type
+## Reading and writing data
 
-^3^: Only lists where all elements are the same type are able to be translated 
-to Arrow list type (which is a "list of" some type).
+One of the main ways to use `arrow` is to read and write data files in
+several common formats. The `arrow` package supplies extremely fast CSV reading and writing capabilities, but in addition supports data formats like Parquet and Feather that are not widely supported in other packages. In addition, the `arrow` package supports multi-file data sets in which a single rectangular data set is stored across multiple files. 
```

Comment:
Okay, what I've done (in order to smooth the transition) is subtly make "Arrow format" or "Arrow IPC format" the primary terminology everywhere, but make sure it always refers to "feather" in a secondary position, either by writing "Arrow/Feather format" or "Arrow (also known as Feather)". It's kind of a soft-deprecation strategy: for the time being I want the word "feather" to be present every time the format is referred to so that people are 100% certain that the Arrow format and the Feather format are the same thing. There's enough confusion on that point already, I think. But once it becomes standard to refer to Arrow format we can probably start deprecating further in the docs?

See what you think!

Reply:
lol, I have to stop using "see what you think" as my way of saying "this is what I think we should do but happy to pivot if you think we need to" 😁

Reply:
Let's loop this into the wider discussion on [ARROW-18148](https://issues.apache.org/jira/browse/ARROW-18148) as I think this is something we need to agree on at a high-level (as it's also being discussed on the mailing list), but I fully agree that some sort of soft-deprecation strategy is needed so we don't confuse users.  I'm thinking "formerly known as" could also be an option, but let's discuss more widely first.